packages/features/ee/billing/api/webhook/_checkout.session.completed.ts

```diff
@@ -64,4 +74,109 @@ async function saveToCreditBalance({
     });
   }
 }
+
+async function handlePhoneNumberSubscription(session: any) {
+  const userId = session.metadata?.userId ? parseInt(session.metadata.userId, 10) : null;
+  const teamId = session.metadata?.teamId ? parseInt(session.metadata.teamId, 10) : null;
+  const agentId = session.metadata?.agentId || null;
+
+  if (!userId || !session.subscription) {
+    console.error("Missing required data for phone number subscription", {
+      userId,
+      hasSubscription: !!session.subscription,
+    });
+    throw new HttpCode(400, "Missing required data for phone number subscription");
+  }
+
+  if (!agentId || agentId?.trim() === "") {
+    console.error("Missing agentId for phone number subscription", {
+      userId,
+      teamId,
+    });
+    throw new HttpCode(400, "Missing agentId for phone number subscription");
+  }
+
+  const agent = await PrismaAgentRepository.findByIdWithUserAccess({
+    agentId,
+    userId,
+    teamId: teamId ?? undefined,
+  });
+
+  if (!agent) {
+    console.error("Agent not found or user does not have access", { agentId, userId });
+    throw new HttpCode(404, "Agent not found or user does not have access to it");
+  }
+
+  const aiService = createDefaultAIPhoneServiceProvider();
+
+  const retellPhoneNumber = await aiService.createPhoneNumber({ nickname: `${userId}-${Date.now()}` });
```

Comment:
We shouldn't have the provider name in the application logic.

it should be calAiPhoneNumber everywhere.